care/facility/models/patient.py

```diff
@@ -806,13 +807,28 @@ class PatientNotes(FacilityBaseModel, ConsultationRelatedPermissionMixin):
         related_name="replies",
     )
     note = models.TextField(default="", blank=True)
+    root_note = models.ForeignKey(
+        "self",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="child_notes",
+    )
 
     def get_related_consultation(self):
         # This is a temporary hack! this model does not have `assigned_to` field
         # and hence the permission mixin will fail if edit/object_read permissions are checked (although not used as of now)
         # Remove once patient notes is made consultation specific.
         return self
 
+    @property
+    def mentioned_users(self):
+        if not hasattr(self, "_mentioned_users_cache"):
+            # handling both - and _ (valid usernames : devdoctor, dev-doctor and dev_doctor)
+            usernames = set(re.findall(r"@([a-zA-Z0-9][-_a-zA-Z0-9]{0,38})", self.note))
+            self._mentioned_users_cache = User.objects.filter(username__in=usernames)
+        return self._mentioned_users_cache
```

Comment:
thoughts on using django's caching framework here instead?

the cache here would only be instance specific right?

cc: @sainak @vigneshhari

Reply:
done, switched to django cache